mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+        """
+        res, response = self._get("restapi/chained_requests/flow/%s/force" % (prepid))
+        return res.get("results", None), response
+
+    def reset(self, prepid) -> tuple[dict | None, Response]:
+        """
+        Reset a request
+        """
+        res, response = self._get("restapi/requests/reset/%s" % (prepid))
+        return res.get("results", None), response
+
+    def soft_reset(self, prepid) -> tuple[dict | None, Response]:
+        """
+        Soft reset a request
+        """
+        res, response = self._get("restapi/requests/soft_reset/%s" % (prepid))
+        return res.get("results", None), response
+
+    def option_reset(self, prepid) -> tuple[dict | None, Response]:
+        """
+        Option reset a request
+        """
+        res, response = self._get("restapi/requests/option_reset/%s" % (prepid))
+        return res.get("results", None), response
+
+    def ticket_generate(self, ticket_prepid) -> tuple[dict | None, Response]:
+        """
+        Generate chains for a ticket
+        """
+        res, response = self._get("restapi/mccms/generate/%s" % (ticket_prepid))
+        return res.get("results", None), response
+
+    def ticket_generate_reserve(self, ticket_prepid) -> tuple[dict | None, Response]:
+        """
+        Generate and reserve chains for a ticket
+        """
+        res, response = self._get("restapi/mccms/generate/%s/reserve" % (ticket_prepid))
+        return res.get("results", None), response
+
+    def rewind(self, chained_request_prepid) -> tuple[dict | None, Response]:
+        """
+        Rewind a chained request
+        """
+        res, response = self._get(
+            "restapi/chained_requests/rewind/%s" % (chained_request_prepid)
+        )
+        return res.get("results", None), response
+
+    def flow(self, chained_request_prepid) -> tuple[dict | None, Response]:
+        """
+        Flow a chained request
+        """
+        res, response = self._get(
+            "restapi/chained_requests/flow/%s" % (chained_request_prepid)
+        )
+        return res.get("results", None), response
+
+    def root_requests_from_ticket(self, ticket_prepid) -> tuple[dict | None, Response]:
+        """
+        Return list of all root (first ones in the chain) requests of a ticket
+        """
+        mccm = self.get("mccms", ticket_prepid)
+        query = ""
+        for root_request in mccm.get("requests", []):
+            if isinstance(root_request, str):
+                query += "%s\n" % (root_request)
+            elif isinstance(root_request, list):
+                # List always contains two elements - start and end of a range
+                query += "%s -> %s\n" % (root_request[0], root_request[1])
+            else:
+                self.logger.error(
+                    "%s is of unsupported type %s", root_request, type(root_request)
+                )
+
+        requests = self.get_range_of_requests(query)
+        return requests
```

Comment:
I would simply drop this part of the "API", it does very little.

Reply:
Agree, already solved